components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/sequence/impl/DefaultStepBasedSequenceHandler.java

```diff
@@ -256,6 +263,11 @@ protected void handlePostAuthentication(HttpServletRequest request,
             }
             ApplicationAuthenticator authenticator = authenticatorConfig.getApplicationAuthenticator();
 
+            String requestedSubject = resolveRequestedSubject(request, context);
```

Comment:
Add a comment explaining what is being done here as we are using the requestedSubject to infer whether this should have a impersonated user details.

Reply:
Fixed with 90f1cbae87d00e5365c738c97476c750c919dd22